File: server/src/test/java/com/ppcomp/server/ServerApplicationTests.java
package com.ppcomp.server;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.web.client.TestRestTemplate;
import org.springframework.boot.web.server.LocalServerPort;
import org.springframework.http.ResponseEntity;

import java.net.URL;

import static org.assertj.core.api.Assertions.assertThat;

@SpringBootTest
class ServerApplicationTests {

	@LocalServerPort
	private int port;

	private URL base;

	@Autowired
	private TestRestTemplate template;

	@BeforeEach
	public void setUp() throws Exception {
		this.base = new URL("http://localhost:" + port + "/");
	}

	@Test
	public void loadContext() throws Exception {
		ResponseEntity<String> response = template.getForEntity(base.toString(),
				String.class);
		assertThat(response.getBody()).isEqualTo("Greetings from Spring Boot!");
	}
}

File: server/src/test/kotlin/com/ppcomp/server/ServerApplicationTests.kt
package com.ppcomp.server

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.web.client.TestRestTemplate
import org.springframework.boot.web.server.LocalServerPort
import org.springframework.http.ResponseEntity
import java.net.URL


@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class HelloControllerIT {
	@LocalServerPort
	private val port = 0
	private var base: URL? = null

	@Autowired
	private lateinit var template: TestRestTemplate

	@BeforeEach
	@Throws(Exception::class)
	fun setUp() {
		base = URL("http://localhost:$port/")
	}

	@get:Throws(Exception::class)
	@get:Test
	val hello: Unit
		get() {
			val response: ResponseEntity<String> = template.getForEntity(
					base.toString(), String::class.java)
			assertThat(response.body).isEqualTo("Greetings from Spring Boot!" )
		}
}